source/fe/mapping_isoparametric.cc

```diff
@@ -0,0 +1,2238 @@
+              if (dim > 2)
+                {
+                  unit_tangentials[GeometryInfo<dim>::faces_per_cell + i]
+                    .resize(n_original_q_points);
+                  std::fill(
+                    unit_tangentials[GeometryInfo<dim>::faces_per_cell + i]
+                      .begin(),
+                    unit_tangentials[GeometryInfo<dim>::faces_per_cell + i]
+                      .end(),
+                    GeometryInfo<dim>::unit_tangential_vectors[i][1]);
+                }
+            }
+        }
+      else if (this->fe.reference_cell_type() == ReferenceCell::Type::Tri)
+        {
+          Tensor<1, dim> t1;
+          t1[0] = 1;
+          t1[1] = 0;
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[0].emplace_back(t1);
+          t1[0] = -0.70710678118;
+          t1[1] = 0.70710678118;
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[1].emplace_back(t1);
+          t1[0] = 0;
+          t1[1] = -1;
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[2].emplace_back(t1);
+        }
+      else if (this->fe.reference_cell_type() == ReferenceCell::Type::Tet)
+        {
+          Tensor<1, dim> t1;
+
+          t1[0] = 0;
+          t1[1] = 1;
+          t1[2] = 0; // face 0
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[0].emplace_back(t1);
+
+          t1[0] = 1;
+          t1[1] = 0;
+          t1[2] = 0; // face 0
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[4].emplace_back(t1);
+
+          t1[0] = 1;
+          t1[1] = 0;
+          t1[2] = 0; // face 1
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[1].emplace_back(t1);
+
+          t1[0] = 0;
+          t1[1] = 0;
+          t1[2] = 1; // face 1
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[5].emplace_back(t1);
+
+          t1[0] = 0;
+          t1[1] = 0;
+          t1[2] = 1; // face 2
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[2].emplace_back(t1);
+
+          t1[0] = 0;
+          t1[1] = 1;
+          t1[2] = 0; // face 2
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[6].emplace_back(t1);
+
+          t1[0] = -0.75983568565;
+          t1[1] = +0.75983568565;
+          t1[2] = +0; // face 3
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[3].emplace_back(t1);
+
+          t1[0] = -0.75983568565;
+          t1[1] = +0;
+          t1[2] = +0.75983568565; // face 3
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[7].emplace_back(t1);
+        }
+      else
+        {
+          Assert(false, ExcNotImplemented());
+        }
```

Comment:
The computation of the tangents on faces of triangles and tetrahedrons are for now here.